independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java

```diff
@@ -513,31 +513,43 @@ private void buildContextPut(String key, Object value) {
         }
     }
 
-    private static Map<DotName, ClassInfo> findQualifiers(IndexView index) {
+    private Map<DotName, ClassInfo> findQualifiers(IndexView index) {
         Map<DotName, ClassInfo> qualifiers = new HashMap<>();
         for (AnnotationInstance qualifier : index.getAnnotations(DotNames.QUALIFIER)) {
-            qualifiers.put(qualifier.target().asClass().name(), qualifier.target().asClass());
+            ClassInfo qualifierClass = qualifier.target().asClass();
+            if (isExcluded(qualifierClass)) {
+                continue;
+            }
+            qualifiers.put(qualifierClass.name(), qualifierClass);
         }
         return qualifiers;
     }
 
-    private static Map<DotName, ClassInfo> findContainerAnnotations(Map<DotName, ClassInfo> annotations, IndexView index) {
+    private Map<DotName, ClassInfo> findContainerAnnotations(Map<DotName, ClassInfo> annotations, IndexView index) {
         Map<DotName, ClassInfo> containerAnnotations = new HashMap<>();
         for (ClassInfo annotation : annotations.values()) {
             AnnotationInstance repeatableMetaAnnotation = annotation.classAnnotation(DotNames.REPEATABLE);
             if (repeatableMetaAnnotation != null) {
                 DotName containerAnnotationName = repeatableMetaAnnotation.value().asClass().name();
-                containerAnnotations.put(containerAnnotationName, getClassByName(index, containerAnnotationName));
+                ClassInfo containerClass = getClassByName(index, containerAnnotationName);
+                if (isExcluded(containerClass)) {
+                    continue;
```

Comment:
Container annotations are not bindings/qualifiers themselves. This should be left untouched.